File: output/llama3.1-8b/avatar/Python/Java/codeforces_30_A.java
import java.util.Scanner;

public class codeforces_30_A {
    public static double task4(double A, double B, int n) {
        for (int X = -1000; X <= 1000; X++) {
            if (Math.pow(X, n) == B / A) {
                return X;
            }
        }
        return "No solution";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        String[] values = input.split(" ");
        double A = Double.parseDouble(values[0]);
        double B = Double.parseDouble(values[1]);
        int n = Integer.parseInt(values[2]);
        System.out.println(task4(A, B, n));
    }
}

File: output/llama3.1-8b/avatar/Java/Python/codeforces_30_A.py
import math

def solve(sc):
    A = int(sc.next())
    B = int(sc.next())
    n = int(sc.next())

    if A == 0:
        if B == 0:
            print(1)
        else:
            print("No solution")
        return

    if B % A != 0:
        print("No solution")
        return

    B //= A
    neg = B < 0
    if neg and n % 2 == 0:
        print("No solution")
        return

    if neg:
        B = -B

    for x in range(B + 1):
        if math.pow(x, n) == B:
            print(neg and -x or x)
            return

    print("No solution")

def main():
    sc = input()
    solve(sc)

if __name__ == "__main__":
    main()